lib/Gearman/Spawner/Server.pm

```diff
@@ -32,48 +33,45 @@ sub create {
 # start up a gearmand that exits when its parent process does. returns the
 # address of the listening server and its pid
 sub fork_gearmand {
-    my $class = shift;
-
-    # NB: this relies on Gearman::Spawner::Process allowing use of fork,
+	my $class = shift;
+    
+	# NB: this relies on Gearman::Spawner::Process allowing use of fork,
     # run_periodically, exit_with_parent, and loop as class methods instead of
     # object methods
 
-    # get an unused port from the OS
```

Comment:
I see. I think there may be a simpler way to fix the race condition: don't call `$sock->close` in the parent until after the child has been contacted. Since both this call to IO::Socket::INET->new and the one in Gearman::Server specify `Reuse => 1`, keeping the parent's listener open shouldn't prevent the child from listening, and leaving the parent listener open should prevent another uid from grabbing the recycled port. Would you be able to test that or provide a test case that reliably demonstrates the failure against the current code?